Rubberduck.Parsing/VBA/RubberduckParser.cs

```diff
@@ -613,11 +613,9 @@ private Declaration CreateProjectDeclaration(QualifiedModuleName projectQualifie
 
         private void ResolveReferences(DeclarationFinder finder, VBComponent component, IParseTree tree)
         {
+            Debug.Assert(State.Status == ParserState.ResolvedDeclarations);
+
             var state = _state.GetModuleState(component);
```

Comment:
you should probably remove this one, since it causes some major mayhem down below in the catch-blocks